lib/async/queue.rb

```diff
@@ -38,13 +38,15 @@ def empty?
 			@items.empty?
 		end
 		
-		def enqueue(item)
-			@items.push(item)
+		def enqueue(*items)
```

Comment:
Can you please check if this allocates a new array? I like this interface, but I just want to confirm we aren't duplicating the argument by doing this.

Reply:
Yes, a single new Array is allocated here. My assumption is new array is used as a local variable inside the method.
From what I can see method arguments (`MyObject` instances in the script below) are not duplicated.
Do you think this is a problem?

<hr/>

Here's the script I used to test this:

```ruby
require 'async'
require_relative 'lib/async/queue'
require 'memory'

class MyObject
end

Async do
	array = Array.new(3) { MyObject.new }
	queue = Async::Queue.new

	Memory.report {
		queue.enqueue(*array)
	}.print
end
```

Here are the results:

```
# Memory Profile

- Total Allocated: (7.00 B in 1 allocations)
- Total Retained: (0 B in 0 allocations)

## By Gem (40.00 B in 1 allocations)

- (40.00 B in 1 allocations)    other

## By File (40.00 B in 1 allocations)

- (40.00 B in 1 allocations)    test.rb

## By Location (40.00 B in 1 allocations)

- (40.00 B in 1 allocations)    test.rb:12

## By Class (40.00 B in 1 allocations)

- (40.00 B in 1 allocations)    Array

## Strings By Gem

## Strings By Location
```